cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/PaybackFlow.kt

```diff
@@ -0,0 +1,202 @@
+
+            progressTracker.currentStep = VERIFYING_TRANSACTION
+            // Verify that the transaction is valid.
+            txBuilder.verify(serviceHub)
+
+            progressTracker.currentStep = SIGNING_TRANSACTION
+            val signedTx = serviceHub.signInitialTransaction(txBuilder)
+
+            progressTracker.currentStep = INITIALISE_OTHER_PARTY_FLOW
+            // Send the state to the counterparty, and receive it back with their signature.
+            val lenderSession = initiateFlow(iouState.lender)
+            val bankSession = initiateFlow(bank)
+
+            // send/receive part is just for experiment
+            val confirmPayback = lenderSession
+                    .sendAndReceive<Boolean>("ready for receiving payback?")
+                    .unwrap { it }
+
+            require(confirmPayback) { "Oops, lender failed to be ready" }
+
+            progressTracker.currentStep = PAYBACK_COMMUNICATION
+            lenderSession.send(iouState.value)
+
+            progressTracker.currentStep = GATHERING_SIGS
+            val fullySignedTx = subFlow(CollectSignaturesFlow(
+                    signedTx,
+                    setOf(lenderSession, bankSession),
+                    GATHERING_SIGS.childProgressTracker())
+            )
+
+            progressTracker.currentStep = FINALISING_TRANSACTION
+            // Notarise and record the transaction in both parties' vaults.
+            return subFlow(FinalityFlow(fullySignedTx, setOf(lenderSession, bankSession), FINALISING_TRANSACTION.childProgressTracker()))
+        }
+    }
+
+    @InitiatedBy(Initiator::class)
+    class Acceptor(val borrowerPartySession: FlowSession) : FlowLogic<SignedTransaction>() {
+        @Suspendable
+        override fun call(): SignedTransaction {
+            val bank = serviceHub.bankProvider()
+
+            if (!ourIdentity.equals(bank)) {
+                borrowerPartySession.receive<String>().unwrap { it }
+                //send confirmation about being ready to receive payback
+                borrowerPartySession.send(true)
+
+                // just exercise
+                val receivedPayback: Int = borrowerPartySession.receive<Int>().unwrap { it }
+            }
+
+            val signTransactionFlow = object : SignTransactionFlow(borrowerPartySession) {
+                override fun checkTransaction(stx: SignedTransaction) = requireThat {
+                    val allInputStates = stx.tx.inputs.getStxStates(serviceHub)
+
+                    val inIOUState = allInputStates.filterByState<IOUState>().single()
+                    val inCashStates = allInputStates.filterByState<CashState>()
+
+                    val outCashStates = stx.tx.outputsOfType<CashState>()
+                    val lenderPaybackCash = outCashStates
+                            .stream().filter { it.owner.equals(inIOUState.lender) }
+                            .toList()
+
+                    val inputCashSum = inCashStates.sumByLongSecure { it.value }
```

Comment:
The fact that the sums in input and output match is taken care of by the contract. This is not really necessary to check in the flow.

Reply:
I need this sum only to calculate if change for borrower is valid.

Reply:
But the lender does not care if the change for the borrower is valid, right?